frappe/public/js/frappe/form/layout.js

```diff
@@ -140,7 +140,9 @@ frappe.ui.form.Layout = class Layout {
 				fieldtype: "Tab Break",
 				fieldname: "__details",
 			};
-			let first_tab = this.fields[1].fieldtype === "Tab Break" ? this.fields[1] : null;
+
+			let first_field_visible = this.fields.find(element => element.hidden == false);
```

Comment:
Note for reviewer: this works instead of `fields[1]` because `fields[0]` (`__newname`) is also hidden.